frontend/src/Pages/Auth.js

```diff
@@ -0,0 +1,242 @@
+import { useEffect, useState } from "react";
+import { useLocation, useNavigate } from "react-router-dom";
+import { Link } from "react-router-dom";
+import { useAuth } from "../Contexts/authContext";
+import "../Styles/Auth.scss";
+
+const Auth = () => {
+  const [loginData, setLoginData] = useState({ email: "", password: "" });
+  const [signupData, setSignupData] = useState({
+    email: "",
+    firstName: "",
+    lastName: "",
+    password: "",
+    confirmPassword: "",
+  });
+  const [authError, setAuthError] = useState("");
+
+  const location = useLocation();
+  const navigate = useNavigate();
+  const { userState, userDispatch, authenticateUser } = useAuth();
+  const { loading, error, userInfo } = userState;
+
+  const redirect = location.search ? location.search.split("=")[1] : "/";
+
+  const handleLogin = async (e, email, password) => {
+    e.preventDefault();
+    setAuthError("");
+    await authenticateUser(
+      "login",
+      email ? email : loginData.email,
+      password ? password : loginData.password,
+      "null"
+    );
+  };
+
+  const handleSignup = async (e) => {
+    e.preventDefault();
+    setAuthError("");
+
+    if (signupData.password !== signupData.confirmPassword) {
+      setAuthError("Passwords do not match");
+      return;
+    }
+
+    await authenticateUser(
+      "register",
+      signupData.email,
+      signupData.password,
+      `${signupData.firstName} ${signupData.lastName}`
+    );
+  };
+
+  useEffect(() => {
+    const handleTabChange = () => {
+      console.log("att");
+      const tabs = document.querySelectorAll(".tabs h3 a");
```

Comment:
Avoid using querySelector as said by tanay, as best practice

Reply:
Yes its already in my todo list. Just focusing on completing the features now, will refactor the code later on.